ComSemApp/models.py

```diff
@@ -334,11 +334,54 @@ class SpeakingPracticeAttempt(models.Model):
     wpm = models.DecimalField(max_digits=5,decimal_places=2, validators=[MinValueValidator(0)], verbose_name='Words per Minute')
 
     def __str__(self):
-        return f"{self.id}: {self.expression} - {self.correct}% ({self.date.strftime('%d %b %Y')})"
+        return f"{self.pk}: {self.expression} - {self.correct}% ({self.date.strftime('%d %b %Y')})"
 
     class Meta:
         verbose_name = "Speaking Practice Attempt"
 
+class SpeakingPracticeAttemptReviewRequest(models.Model):
+    """
+        This model stores students' requests for teacher review on their Speaking Practice attempts
+        Inherits from:
+            django.db.models.Model
+        Remarks:
+            The 'attempt' field is the primary key for this model because students cannot request
+            review of the same attempt more than once.
+    """
+    # The attempt which the student has requested review for
+    attempt = models.OneToOneField(SpeakingPracticeAttempt, on_delete=models.CASCADE, primary_key=True)
+    # The date on which the request was created
+    date = models.DateField(auto_now_add=True)
+
+    def __str__(self):
+        return f"{self.attempt.pk}: {self.date}"
+    
+    class Meta:
+        verbose_name = "Instructor Review Request for Speaking Practice Attempt"
+
+class SpeakingPracticeAttemptReview(models.Model):
+    """
+        This model stores teachers' reviews of students' SpeakingPracticeAttemptReviewRequests
+        Inherits from:
+            django.db.models.Model
+        Remarks:
+            The 'request' field is the primary key for this model because each attempt can be
+            satisfied by one review
+    """
+    # The SpeakingPracticeAttemptReviewRequest that this review is satisfying
+    request = models.OneToOneField(SpeakingPracticeAttemptReviewRequest, on_delete=models.CASCADE, primary_key=True)
+    # The Teacher that reviewed the attempt
+    reviewer = models.ForeignKey(Teacher, on_delete=models.SET_NULL, null=True)
+    # The date on which the review was completed
+    date = models.DateField(auto_now_add=True)
+    # The comments the teacher has provided during review
+    comments = models.TextField(null=True, blank=True, verbose_name="Teacher Comments")
+    # The original score of the attempt. If blank, the score was not updated
+    original_score = models.DecimalField(null=True, blank=False, max_digits=5, decimal_places=2, validators=[MinValueValidator(0), MaxValueValidator(100)], verbose_name='Original Score')
```

Comment:
I am assuming that we plan on storing the teacher score by overwriting the score in the attempt?

Reply:
Correct. I'll update the comment to reflect the change.